File: Integration/CiteDeLaCultureIntegration/src/main/java/Entite/Inscription.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Entite;

/**
 *
 * @author Elyes
 */
public class Inscription {

    int id_inscription, id_event, id_formation, id_user;

    
    public Inscription(int id_inscription, int id_event, int id_formation, int id_user) {
        this.id_inscription = id_inscription;
        this.id_event = id_event;
        this.id_formation = id_formation;
        this.id_user = id_user;
    }
    
    
    public Inscription(int id_event, int id_formation, int id_user) {
        this.id_formation = id_formation;
        this.id_event = id_event;
        this.id_user = id_user;
    }
    
    

    public Inscription(int id_event, int id_user) {
        this.id_event = id_event;
        this.id_user = id_user;
    }

    public Inscription() {
    }

    
    
    
    
    public int getId_formation() {
        return id_formation;
    }

    public void setId_formation(int id_formation) {
        this.id_formation = id_formation;
    }
    

    public int getId_inscription() {
        return id_inscription;
    }

    public void setId_inscription(int id_inscription) {
        this.id_inscription = id_inscription;
    }

    public int getId_event() {
        return id_event;
    }

    public void setId_event(int id_event) {
        this.id_event = id_event;
    }

   

    public int getId_user() {
        return id_user;
    }

    public void setId_user(int id_user) {
        this.id_user = id_user;
    }

    @Override
    public String toString() {
        return "Inscription{" + "id_inscription=" + id_inscription + ", id_event=" + id_event + ", id_formation=" + id_formation + ", id_user=" + id_user + '}';
    }

    
    
    
    
    

}

File: src/WebBundle/Entity/Inscription.php
<?php

namespace WebBundle\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * Inscription
 *
 * @ORM\Table(name="inscription", indexes={@ORM\Index(name="fk", columns={"iduser"}), @ORM\Index(name="fk1", columns={"idformation"}), @ORM\Index(name="fk2", columns={"idevent"})})
 * @ORM\Entity(repositoryClass="WebBundle\Repository\InscriptionRepository")
 */
class Inscription
{
    /**
     * @var integer
     *
     * @ORM\Column(name="idinscription", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $idinscription;

    /**
     * @var \Evenement
     *
     * @ORM\ManyToOne(targetEntity="Evenement")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="idevent", referencedColumnName="idevent")
     * })
     */
    private $idevent;

    /**
     * @var \Formation
     *
     * @ORM\ManyToOne(targetEntity="Formation")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="idformation",  referencedColumnName="idformation")
     * })
     */
    private $idformation;

    /**
     * @var \User
     *
     * @ORM\ManyToOne(targetEntity="User")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="iduser", referencedColumnName="iduser")
     * })
     */
    private $iduser;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="dateAjout", type="datetime", nullable=true)
     */
    private $dateajout = 'CURRENT_TIMESTAMP';

    public function __construct() {
        $this->dateajout = new \DateTime();
    }

    /**
     * @return int
     */
    public function getIdinscription()
    {
        return $this->idinscription;
    }

    /**
     * @param int $idinscription
     */
    public function setIdinscription($idinscription)
    {
        $this->idinscription = $idinscription;
    }

    /**
     * @return \Evenement
     */
    public function getIdevent()
    {
        return $this->idevent;
    }

    /**
     * @param \Evenement $idevent
     */
    public function setIdevent($idevent)
    {
        $this->idevent = $idevent;
    }

    /**
     * @return \Formation
     */
    public function getIdformation()
    {
        return $this->idformation;
    }



    /**
     * @param \Formation $idformation
     */
    public function setIdformation($idformation)
    {
        $this->idformation = $idformation;
    }

    /**
     * @return \User
     */
    public function getIduser()
    {
        return $this->iduser;
    }

    /**
     * @param \User $iduser
     */
    public function setIduser($iduser)
    {
        $this->iduser = $iduser;
    }

    /**
     * @return \DateTime
     */
    public function getDateajout()
    {
        return $this->dateajout;
    }

    /**
     * @param \DateTime $dateajout
     */
    public function setDateajout($dateajout)
    {
        $this->dateajout = $dateajout;
    }












}

